oauth2-server-core/src/main/java/nl/myndocs/oauth2/grant/CallRouterRedirect.kt

```diff
@@ -13,54 +11,39 @@ import nl.myndocs.oauth2.scope.ScopeParser
 import nl.myndocs.oauth2.token.AccessToken
 import nl.myndocs.oauth2.token.CodeToken
 
-
 fun GrantingCall.redirect(redirect: RedirectAuthorizationCodeRequest): CodeToken {
-    if (redirect.clientId == null) {
-        throw InvalidRequestException(INVALID_REQUEST_FIELD_MESSAGE.format("client_id"))
-    }
-
-    if (redirect.username == null) {
-        throw InvalidRequestException(INVALID_REQUEST_FIELD_MESSAGE.format("username"))
-    }
-
-    if (redirect.password == null) {
-        throw InvalidRequestException(INVALID_REQUEST_FIELD_MESSAGE.format("password"))
-    }
-    if (redirect.redirectUri == null) {
-        throw InvalidRequestException(INVALID_REQUEST_FIELD_MESSAGE.format("redirect_uri"))
-    }
-
-    val clientOf = clientService.clientOf(redirect.clientId) ?: throw InvalidClientException()
+    checkMissingFields(redirect)
 
+    val clientOf = clientService.clientOf(redirect.clientId!!) ?: throw InvalidClientException()
     if (!clientOf.redirectUris.contains(redirect.redirectUri)) {
         throw InvalidGrantException("invalid 'redirect_uri'")
     }
 
-    val authorizedGrantType = AuthorizedGrantType.AUTHORIZATION_CODE
-    if (!clientOf.authorizedGrantTypes.contains(authorizedGrantType)) {
-        throw InvalidGrantException("Authorize not allowed: '$authorizedGrantType'")
+    with(AuthorizedGrantType.AUTHORIZATION_CODE) {
```

Comment:
In this case, I don't see what `with` provides more then previous solution?
It requires extra level of nesting, and instead of assigning it to a named variable, it is now assigned to `this`

Reply:
I thought this extra level of nesting would be clearer than introducing a new variable that will be used only once. the nested block isn't big enough though. So, I guess it's between the gap of noisy and clean code, or it's just a matter of preference.